pynamodb/connection/base.py

```diff
@@ -365,6 +365,8 @@ def _make_api_call(self, operation_name, operation_kwargs):
             except (requests.RequestException, ValueError) as e:
                 if is_last_attempt_for_exceptions:
                     log.debug('Reached the maximum number of retry attempts: %s', attempt_number)
+                    if response is not None:
+                        response.raise_for_status()
```

Comment:
The two lines below (370-371) are effectively unreachable with this. How about just changing it to:

```
if response:
  response.raise_for_status()
raise
```